### ISSUE TITLE
Log Out Manually

### ISSUE THREAD

**Issue by kapilpundit** [rmartide/vue-google-login#3]
Hi @rmartide 
Thanks for this nice plugin.

I would appreciate you if you can let me know how I can manually log out from google after logging in.

In my Vue app, I log the user in using vue-google-login, but, then after collecting the information, I would log the user out of google.

I am using vue-router, so the login page is on /login. I have already tried LoaderPlugin but could not successfully call it as below:
`GoogleAuth.then(auth2 => {
            auth2.signOut();
        })`
or even 
`this.GoogleAuth.then(auth2 => {
            auth2.signOut();
        })`

or even
`this.$GoogleAuth.then(auth2 => {
            auth2.signOut();
        })`

How can this be made possible.

Thanks,

**Comment by rmartide** [comment]
Hi.

You are right, for some reason the promise is not resolving.

I'm checking and it's working before the build but not after, it's quite strange, I assumed it would work since the login does.

I'm trying to figure out how to fix it.

I'll post again later when I update the plugin with the fix.

Cheers.

**Comment by rmartide** [comment]
I edited the previous comment, I'll post this just in case you don't get an update.

**Comment by kapilpundit** [comment]
Hi @rmartide 
Thanks for your reply, and sorry I got busy somewhere so could not revert on time.

I tried your code, but somehow I don't get logged out of google from the browser.
`auth2.isSignedIn.get()`
returns true two times.

I even tried this
`Vue.GoogleAuth.then(auth2 => {
            // console.log(auth2);
            auth2.signOut().then(() => {
               auth2.disconnect();
              // console.log(auth2.isSignedIn.get());
            });
          });`

Still not getting logged out of google. Am I missing something, or this code won't let me log out of google (I mean I am making a conceptual mistake).
Thanks,

**Comment by rmartide** [comment]
Mmm. I am not sure at this point, the method is from the google api, I guess you checked it too since you tried disconnect.

The api sais about signOut: A Promise that is fulfilled when the user has been signed out.

So I guess that should mean if the promise resolves in the then callback it should log out the user.

When I use the code I pasted you I get true, false.

I wish I could help you more but at this point I'm clueless.

The logout button works? I know you want to manually log out with code but have you tried just to see if it works? I'm curious.

**Comment by kapilpundit** [comment]
I have not tried the logout button. Let me try it and get back to you.

You get true, false when you try the same code. Can it mean that there is either something wrong with the google app I created in the google developers console?

Can you please suggest me which all APIs need to be enabled through google developers console for google login?

**Comment by rmartide** [comment]
I don't think you did anything wrong since Auth2 are just credentials linked to an API.

I know what we can try. Download or fork the repo and try the dev project.

Try the code there. It has already a client id, the same I used to develop the plugin. It is also where I tried the code posted above.

To start it is just npm install and npm start.

The file to change is dev/App.vue, there is the handleClick method for the isLoggedIn button. Uncomment the code and try it.

Let me know how it goes.
